src/granite/settings.cr

```diff
@@ -2,13 +2,25 @@ require "logger"
 
 module Granite
   class Settings
-    property database_url : String? = nil
-    property logger : Logger
+    property logger = Logger.new STDOUT
+    getter adapters = {} of String => String
 
     def initialize
-      @logger = Logger.new STDOUT
       @logger.progname = "Granite"
     end
+
+    def database_url=(database_url : String)
```

Comment:
I see, while this is cool, I don't think we should do this, is confusion, reasons why is this is an odd behavior for an assignment method `=` this would be much better for an append method `<<`. And again 1 instance of Granite should manage 1 connection IMO.

Reply:
Also what if someone plug in a MemSQL adapter to granite and wants to use it, this will not allow it since the method is generalized. :)

Reply:
In most cases only 1 connection_url would be used, Granite itself needs all 3 for the specs, since the specs run for each adapter.

It would error saying that it's an invalid adapter.

Reply:
It seems that the spec should have an instance of Granite per Adapter, not the way is currently set up. So instead of defining a single spec helper with all 3 connections I will define a spec helper per adapter and require it per the corresponding adapter.

Reply:
> EDIT: Granite in the past supported multiple connections, but in the form of the database.yml config file. So it's not like this is anything new.

Thank you for the info, I didn't know this detail.

Reply:
Yea https://github.com/amberframework/granite/blob/master/config/database.yml

So how that worked was, code in Adapter::Base parsed that out and inserted the database_urls from ENV.  This way, when a new instance of an adapter was created from the `adapter` macro, it just picked the correct one.